dataframely/columns/_base.py

```diff
@@ -97,7 +149,26 @@ def validation_rules(self, expr: pl.Expr) -> dict[str, pl.Expr]:
         if not self.nullable:
             result["nullability"] = expr.is_not_null()
         if self.check is not None:
-            result["check"] = self.check(expr)
+            if isinstance(self.check, dict):
+                for rule_name, rule_callable in self.check.items():
+                    result[rule_name] = rule_callable(expr)
+
+            elif isinstance(self.check, list):
+                # Get unique names for rules from callables
+                rule_names = self._get_rule_names(rules=self.check)
+
+                # Apply each rule with its corresponding name
+                for rule_name, rule_callable in zip(rule_names, self.check):
+                    result[rule_name] = rule_callable(expr)
+
+            else:
```

Comment:
[nit] since this is a nested conditional, I rather find that the spacing between the cases makes the code harder to read. Let's remove it

```suggestion
            elif isinstance(self.check, list):
                # Get unique names for rules from callables
                rule_names = self._get_rule_names(rules=self.check)

                # Apply each rule with its corresponding name
                for rule_name, rule_callable in zip(rule_names, self.check):
                    result[rule_name] = rule_callable(expr)
            else:
```